code/game/machinery/computer/cloning.dm

```diff
@@ -568,79 +609,93 @@ DEFINE_BUFFER_HANDLER(/obj/machinery/computer/cloning)
+	var/mob/living/brain/human_brain = mob_occupant
 
 	if(ishuman(mob_occupant))
-		dna = C.has_dna()
-		var/obj/item/card/id/I = C.get_idcard(TRUE)
-		if(I)
-			has_bank_account = I.registered_account
+		dna = human_mob.has_dna()
+		var/obj/item/card/id/human_id_card = human_mob.get_idcard(TRUE)
+		if(human_id_card)
+			has_bank_account = human_id_card.registered_account
 	if(isbrain(mob_occupant))
-		dna = B.stored_dna
+		dna = human_brain.stored_dna
 
 	if(!can_scan(dna, mob_occupant, has_bank_account, body_only))
 		return
 
-	var/datum/data/record/R = new()
+	var/datum/record/cloning/cloning_record = new()
 	if(dna.species)
 		// We store the instance rather than the path, because some
 		// species (abductors, slimepeople) store state in their
 		// species datums
 		dna.delete_species = FALSE
-		R.fields["mrace"] = dna.species
+		cloning_record.species = dna.species
 	else
 		return //no dna info for species? you're not allowed to clone them. Don't harass xeno, don't try xeno farm.
 		//Note: if you want to clone unusual species, you need to check 'carbon/human' rather than 'dna.species'
 
-	R.fields["name"] = mob_occupant.real_name
+	cloning_record.name = mob_occupant.real_name
 	if(experimental) //even if you have the same identity, this will give you different id based on your mind. body_only gets β at their id.
-		R.fields["id"] =  copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 10)+"β+" //beta plus
+		cloning_record.id =  copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 10)+"β+" //beta plus
 	else if(body_only)
-		R.fields["id"] = copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 10)+"β" //beta
+		cloning_record.id = copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 10)+"β" //beta
 	else
-		R.fields["id"] = copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 7)+copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.mind), -4)
-	R.fields["UE"] = dna.unique_enzymes
-	R.fields["UI"] = dna.uni_identity
-	R.fields["SE"] = dna.mutation_index
-	R.fields["blood_type"] = dna.blood_type
-	R.fields["features"] = dna.features
-	R.fields["factions"] = mob_occupant.faction
-	R.fields["traumas"] = list()
+		cloning_record.id = copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.real_name), 3, 7)+copytext_char(rustg_hash_string(RUSTG_HASH_MD5, mob_occupant.mind), -4)
+	cloning_record.UE = dna.unique_enzymes
+	cloning_record.uni_identity = dna.uni_identity
+	cloning_record.SE = dna.mutation_index
+	cloning_record.blood_type = dna.blood_type
+	cloning_record.weakref_dna = WEAKREF(dna)
+	cloning_record.factions = mob_occupant.faction
+	cloning_record.traumas = list()
+	cloning_record.age = human_mob.age
+	cloning_record.dna_string = dna.unique_enzymes
+	cloning_record.fingerprint = md5(dna.uni_identity)
+	cloning_record.gender = human_mob.gender
+	cloning_record.initial_rank = human_mob.mind.assigned_role
```

Comment:
isn't it better with giving values already?
instead of trying `new()` prior, declare variables to store values first, then pass the values to the `new()`

Reply:
I believe i did this correctly? Please do show an example if it is wrong.

Reply:
Slightly missed but yeah you did correctly